src/win/udp.c

```diff
@@ -422,57 +422,71 @@ void uv__process_udp_recv_req(uv_loop_t* loop, uv_udp_t* handle,
     DWORD bytes, err, flags;
     struct sockaddr_storage from;
     int from_len;
+    int count;
+
+    /* Prevent loop starvation when the data comes in as fast as (or faster than)
+     * we can read it.
+     */
+    count = 32;
+
+    do {
+      /* Do at most `count` nonblocking receive. */
+      buf = uv_buf_init(NULL, 0);
+      handle->alloc_cb((uv_handle_t*) handle, UV__UDP_DGRAM_MAXSIZE, &buf);
+      if (buf.base == NULL || buf.len == 0) {
+        handle->recv_cb(handle, UV_ENOBUFS, &buf, NULL, 0);
+        goto done;
+      }
+      assert(buf.base != NULL);
```

Comment:
I realize it's existing code but:
```suggestion
```
Logically impossible because of the preceding `if`.